src/posit/connect/permissions.py

```diff
@@ -67,36 +67,105 @@ def count(self) -> int:
         return len(self.find())
 
     @overload
-    def create(self, *, principal_guid: str, principal_type: str, role: str) -> Permission:
+    def create(self, *, principal_guid: str, principal_type: str, role: str) -> Permission: ...
+
+    @overload
+    def create(self, *args: User | Group, role: str) -> list[Permission]: ...
+
+    def create(self, *args: User | Group, **kwargs) -> Permission | list[Permission]:
```

Comment:
While I understand that changing return type of `create()` to always be `list[Permission]` is not wise from backward compatibility point of view, but I think that having a function switching behaviour and return type based on the arguments can be confusing. Especially in the context of type hinting.

Why not make a `create()` and `create_many()` functions?
`create()` could be a simple wrapper on top of `create_many()` and so it wouldn't add any additional maintenance cost.
At that point each one of the two can have a clear and unique signature and even from autocompletion point of view for a user it will be more obvious to see that they have two choices.

Reply:
Pairing with @schloerke through a review of this, and we're discussing this.

This seems like a really reasonable approach. Question:

Should `create()` adopt user and group support, so you can call `.create(my_user, role="viewer")`?

Reply:
If `.create()` adopted user/group support, I'd be fine not implementing `.create_many()`

```python
    def create(self, principal: User | Group, /, **kwargs: Unpack[CreateArgs]) -> Permission:
```

Reply:
Then you could just use it as in a list comprehension!

Reply:
Settled on :

```python
    @overload
    def create(self, /, *, principal_guid: str, principal_type: str, role: str) -> Permission: ...

    @overload
    def create(self, principal: User | Group, /, *, role: str) -> Permission: ...
```

😃😃